python/setup.py

```diff
@@ -17,6 +17,8 @@
 
 REQUIREMENTS = [
     "numpy",
+    "scipy",
```

Comment:
The SciPy dependency should be removed in DOLFIN - it's really just a bit of testing code. I can be a test dependency.

Reply:
I see the only one reference to scipy (outside of testing code) which is here https://github.com/FEniCS/dolfinx/blob/0c0fbc15367ae3ba1faddc867177bf38a8ef56e2/python/dolfinx/fem/assemble.py#L196

Reply:
I think it is also in pybind implicitly (pybind understands scipy sparse <-> Eigen sparse translation).